src/main/java/org/folio/rest/impl/LoansAPI.java

```diff
@@ -180,6 +186,63 @@ public void postLoanStorageLoansAnonymizeByUserId(
     });
   }
 
+  @Override
+  public void postLoanStorageAnonymizeloans(List<String> loanIds,
+                                            Map<String, String> okapiHeaders,
+                                            Handler<AsyncResult<Response>> responseHandler,
+                                            Context vertxContext) {
+
+    final ServerErrorResponder serverErrorResponder =
+      new ServerErrorResponder(PostLoanStorageLoansAnonymizeByUserIdResponse
+        ::respond500WithTextPlain, responseHandler, log);
+
+    final VertxContextRunner runner = new VertxContextRunner(
+      vertxContext, serverErrorResponder::withError);
+    AnonymizeLoansResponse response = new AnonymizeLoansResponse();
+    runner.runOnContext(() -> {
+      final Set<String> invalidUuids = loanIds.stream()
+        .filter(id -> !UUIDValidation.isValidUUID(id))
+        .collect(Collectors.toSet());
+
+      if (!invalidUuids.isEmpty()) {
+        final Errors errors = ValidationHelper.createValidationErrorMessage(
+          "invalidLoanIds",
+          Json.encode(invalidUuids),
+          "Loan IDs should be a valid UUID");
+        responseHandler.handle(succeededFuture(
+          PostLoanStorageLoansAnonymizeByUserIdResponse
+            .respond422WithApplicationJson(errors)));
+        return;
+      }
+
+      final String tenantId = TenantTool.tenantId(okapiHeaders);
+      final PostgresClient postgresClient = PostgresClient.getInstance(
+        vertxContext.owner(), tenantId);
+
+      final String combinedAnonymizationSql = createAnonymizationSQL(loanIds,
+        tenantId);
+      log.info(String.format("Anonymization SQL: %s", combinedAnonymizationSql));
+      update(postgresClient, combinedAnonymizationSql)
+        .map(updateResult -> {
+          response.setUpdatedCount(updateResult.getUpdated());
```

Comment:
`response.withUpdatedCount` can be used here to shorten the lambda expression to one line

Reply:
Done.